fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/SplitScanORToUnionRule.java

```diff
@@ -0,0 +1,194 @@
+import com.starrocks.sql.optimizer.Utils;
+import com.starrocks.sql.optimizer.operator.OperatorType;
+import com.starrocks.sql.optimizer.operator.logical.LogicalOlapScanOperator;
+import com.starrocks.sql.optimizer.operator.logical.LogicalUnionOperator;
+import com.starrocks.sql.optimizer.operator.pattern.Pattern;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.rewrite.scalar.FilterSelectivityEvaluator;
+import com.starrocks.sql.optimizer.rewrite.scalar.FilterSelectivityEvaluator.ColumnFilter;
+import com.starrocks.sql.optimizer.rewrite.scalar.NegateFilterShuttle;
+import com.starrocks.sql.optimizer.rule.RuleType;
+import com.starrocks.sql.optimizer.statistics.Statistics;
+import com.starrocks.sql.optimizer.statistics.StatisticsCalcUtils;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SplitScanORToUnionRule extends TransformationRule {
+    private static final Logger LOG = LogManager.getLogger(SplitScanORToUnionRule.class);
+    private static final double SELECTIVITY_THRESHOLD = 0.035;
+
+    private SplitScanORToUnionRule() {
+        super(RuleType.TF_SPLIT_SCAN_OR, Pattern.create(OperatorType.LOGICAL_OLAP_SCAN));
+    }
+
+    public static SplitScanORToUnionRule getInstance() {
+        return INSTANCE;
+    }
+
+    private static final SplitScanORToUnionRule INSTANCE = new SplitScanORToUnionRule();
+
+    @Override
+    public boolean check(final OptExpression input, OptimizerContext context) {
+        LogicalOlapScanOperator scan = (LogicalOlapScanOperator) input.getOp();
+        return context.getSessionVariable().getScanOrToUnionLimit() > 1 && scan.getPredicate() != null
+                && !scan.isFromSplitOR();
+    }
+
+
+
+    @Override
+    public List<OptExpression> transform(OptExpression input, OptimizerContext context) {
+        try {
+            return transformImpl(input, context);
+        } catch (Exception e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("input: {}, msg: {}", input.explain(), e.getMessage());
+            }
+            return Lists.newArrayList();
+        }
+    }
+
+    private List<OptExpression> transformImpl(OptExpression input, OptimizerContext context) {
+        LogicalOlapScanOperator scan = (LogicalOlapScanOperator) input.getOp();
+        long totalRowCount = StatisticsCalcUtils.getTableRowCount(scan.getTable(), scan);
+        Statistics.Builder builder = StatisticsCalcUtils.estimateScanColumns(scan.getTable(),
+                scan.getColRefToColumnMetaMap());
+        Statistics statistics = builder.setOutputRowCount(totalRowCount).build();
+
+        if (statistics.getComputeSize() <= context.getSessionVariable().getScanOrToUnionThreshold()) {
```

Comment:
should consider no statistics case

Reply:
The `FilterSelectivityEvaluator` sets the filter to `NON_SELECTIVITY` when statistic is unknown, so no need to process it here.

Reply:
Further confirm, what you mean is, if no statistics, will we spilt the or predicate to union?